File: eio2024-lv/2024-10-07-lv/vandra/solution/sol_naive.py
# korrektne, aga ebaefektiive lahendus: vaatab läbi kõik laudade paarid
# ja kontrollib iga paari kohta, kas need kaks lauda omavahel kattuvad

# lauda esitab nelik [x1, y1, x2, y2]
def kattuvad(l1, l2):
	if l1[0] >= l2[2]:
		# l1 tervenisti paremal
		return False
	if l2[0] >= l1[2]:
		# l2 tervenisti paremal
		return False
	if l1[1] >= l2[3]:
		# l1 tervenisti kõrgemal
		return False
	if l2[1] >= l1[3]:
		# l2 tervenisti kõrgemal
		return False
	return True

# loeme laudade andmed
n = int(input())
lauad = []
for i in range(n):
    lauad.append([int(s) for s in input().split()])

# leiame vastuse
vastus = False
for i in range(n):
	for j in range(i):
		if kattuvad(lauad[i], lauad[j]):
			vastus = True

# väljastame vastuse
if vastus:
    print("JAH")
else:
    print("EI")

File: eio2024-lv/2024-10-07-lv/vandra/solution/sol_naive.cpp
// korrektne, aga ebaefektiive lahendus: vaatab läbi kõik laudade paarid
// ja kontrollib iga paari kohta, kas need kaks lauda omavahel kattuvad

#include <iostream>
#include <vector>
using namespace std;

struct laud {
	int x1, y1, x2, y2;
};

bool kattuvad(const laud & l1, const laud & l2) {
	if (l1.x1 >= l2.x2) {
		// l1 tervenisti paremal
		return false;
	}
	if (l2.x1 >= l1.x2) {
		// l2 tervenisti paremal
		return false;
	}
	if (l1.y1 >= l2.y2) {
		// l1 tervenisti kõrgemal
		return false;
	}
	if (l2.y1 >= l1.y2) {
		// l2 tervenisti kõrgemal
		return false;
	}
	return true;
}

int main() {
	// loeme laudade andmed
	int n;
	cin >> n;
	vector<laud> lauad(n);
	for (auto & l : lauad) {
		cin >> l.x1 >> l.y1 >> l.x2 >> l.y2;
	}

	// leiame vastuse
	bool vastus = false;
	for (int i = 0; i < n; ++i) {
		for (int j = 0; j < i; ++j) {
			if (kattuvad(lauad[i], lauad[j])) {
				vastus = true;
			}
		}
	}

	// väljastame vastuse
	if (vastus) {
		cout << "JAH" << '\n';
	} else {
		cout << "EI" << '\n';
	}
}
